src/hotspot/cpu/aarch64/aarch64_sve.ad

```diff
@@ -2684,3 +2684,135 @@ instruct vcvtLtoB(vReg dst, vReg src, vReg tmp)
+  match(Set dst (VectorCastB2X src));
+  ins_cost(3 * SVE_COST);
+  format %{ "sve_sunpklo  $dst, H, $src\n\t"
+            "sve_sunpklo  $dst, S, $dst\n\t"
+            "sve_scvtf  $dst, S, $dst, S\t# convert B to F vector" %}
+  ins_encode %{
+    __ sve_sunpklo(as_FloatRegister($dst$$reg), __ H, as_FloatRegister($src$$reg));
+    __ sve_sunpklo(as_FloatRegister($dst$$reg), __ S, as_FloatRegister($dst$$reg));
+    __ sve_scvtf(as_FloatRegister($dst$$reg), __ S, ptrue, as_FloatRegister($dst$$reg), __ S);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vcvtStoD(vReg dst, vReg src)
+%{
+  predicate(UseSVE > 0 && n->bottom_type()->is_vect()->length_in_bytes() >= 16 &&
+            n->bottom_type()->is_vect()->element_basic_type() == T_DOUBLE);
+  match(Set dst (VectorCastS2X src));
+  ins_cost(3 * SVE_COST);
+  format %{ "sve_sunpklo  $dst, S, $src\n\t"
+            "sve_sunpklo  $dst, D, $dst\n\t"
+            "sve_scvtf  $dst, D, $dst, D\t# convert S to D vector" %}
+  ins_encode %{
+    __ sve_sunpklo(as_FloatRegister($dst$$reg), __ S, as_FloatRegister($src$$reg));
+    __ sve_sunpklo(as_FloatRegister($dst$$reg), __ D, as_FloatRegister($dst$$reg));
+    __ sve_scvtf(as_FloatRegister($dst$$reg), __ D, ptrue, as_FloatRegister($dst$$reg), __ D);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+
+instruct vcvtBtoD(vReg dst, vReg src)
+%{
+  predicate(UseSVE > 0 && n->bottom_type()->is_vect()->length_in_bytes() >= 16 &&
+            n->bottom_type()->is_vect()->element_basic_type() == T_DOUBLE);
+  match(Set dst (VectorCastB2X src));
+  effect(TEMP_DEF dst);
+  ins_cost(3 * SVE_COST);
+  format %{ "sve_sunpklo  $dst, H, $src\n\t"
+            "sve_sunpklo  $dst, S, $dst\n\t"
+            "sve_sunpklo  $dst, D, $dst\n\t"
+            "sve_D  $dst, scvtf, $dst, scvtf\t# convert B to D vector" %}
+  ins_encode %{
+    __ sve_sunpklo(as_FloatRegister($dst$$reg), __ H, as_FloatRegister($src$$reg));
+    __ sve_sunpklo(as_FloatRegister($dst$$reg), __ S, as_FloatRegister($dst$$reg));
+    __ sve_sunpklo(as_FloatRegister($dst$$reg), __ D, as_FloatRegister($dst$$reg));
+    __ sve_scvtf(as_FloatRegister($dst$$reg), __ D, ptrue, as_FloatRegister($dst$$reg), __ D);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+
+instruct vcvtLtoF(vReg dst, vReg src, vReg tmp)
+%{
+  predicate(UseSVE > 0 && n->bottom_type()->is_vect()->length_in_bytes() >= 16 &&
+            n->bottom_type()->is_vect()->element_basic_type() == T_FLOAT);
+  match(Set dst (VectorCastL2X src));
+  effect(TEMP_DEF dst, TEMP tmp);
+  ins_cost(3 * SVE_COST);
+  format %{ "sve_scvtf  $dst, S, $src, D\n\t"
+            "sve_dup  $tmp, S, 0\n\t"
+            "sve_uzp1  $dst, S, $dst, $tmp\t# convert L to F vector" %}
+  ins_encode %{
+    __ sve_scvtf(as_FloatRegister($dst$$reg), __ S, ptrue, as_FloatRegister($src$$reg), __ D);
+    __ sve_dup(as_FloatRegister($tmp$$reg), __ S, 0);
+    __ sve_uzp1(as_FloatRegister($dst$$reg), __ S, as_FloatRegister($dst$$reg), as_FloatRegister($tmp$$reg));
```

Comment:
This looks an issue to me. From the encoding, the `"__ sve_scvtf(as_FloatRegister($dst$$reg), __ S, ptrue, as_FloatRegister($src$$reg), __ D);"` can directly converts 64-bits to single-precision. Why do you need the followed `sve_uzp1` instruction? I guess you want convert "long" to "double" and then to "float" ? If so, the first `scvtf` instruction should be `"__ sve_scvtf(as_FloatRegister($dst$$reg), __ D, ptrue, as_FloatRegister($src$$reg), __ D);"`, and then encoding an `fcvt` which converts `"double"` to `"float"`. Please correct me if I'm wrong. Thanks!